lib/qesapdeployment.pm

```diff
@@ -1707,4 +1730,86 @@ sub qesap_export_instances {
     record_info('EXPORT', "SSH keys and instances data uploaded to test results:\n" . join("\n", @upload_files));
 }
 
+=head3 get_job_state
+
+    Get whether a specified job is still running or not
+
+=over 1
+
+=item B<JOB_ID> - id of job to check
+
+=back
+=cut
+
+sub get_job_state {
+    my ($job_id) = @_;
+
+    my $url = "https://openqa.suse.de/api/v1/jobs/$job_id";
+
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;    # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    if ($job_state eq 'running') {
+        return 0;    # Job is still running
+    }
+    else {
+        return 1;    # Job has stopped
+    }
+}
+
+
+=head3 qesap_az_get_active_peerings
+
+    Get active peerings for Azure jobs
+
+=over 2
+
+=item B<RG> - Resource group in question
+
+=item B<VNET> - vnet name of rg
+
+=back
+=cut
+
+sub qesap_az_get_active_peerings {
+    my (%args) = @_;
+    foreach (qw(rg vnet)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my $cmd = "az network vnet peering list -g $args{rg} --vnet-name $args{vnet} --query \"[].name\"";
+    my $output_str = script_output($cmd);
+    my @output = split(/\n/, $output_str);
+    my %result;
+
+    foreach my $line (@output) {
+        my @matches = $line =~ /(\d{6,})/g;
+        $result{$line} = $matches[-1] if @matches;
+    }
+    return %result;
+}
+
+=head2 qesap_az_clean_old_peerings
+
+    Delete leftover peerings for Azure jobs that finished without cleaning up
+
+=cut
+
+sub qesap_az_clean_old_peerings {
+    my (%args) = @_;
+    foreach (qw(rg vnet)) { croak "Missing mandatory $_ argument" unless $args{$_}; }
+    my %peerings = qesap_az_get_active_peerings(rg => $args{rg}, vnet => $args{vnet});
+
+    while (my ($key, $value) = each %peerings) {
+        if (get_job_state($value) == 1) {
```

Comment:
No need to check agains `== 1`. Suggestion:
```
if (get_job_state($value)) {
```
In fact, since `get_job_state()` returns 1 or 0, it's not actually returning the job state, but telling you whether it's running or not. I'd rename the function to `is_job_finished` (or something along those lines), and then modify this line to:
```
if (is_job_finished($value)) {
```

Reply:
Done